examples/perplexity/perplexity.cpp

```diff
@@ -412,13 +412,22 @@ static results_perplexity perplexity_v2(llama_context * ctx, const common_params
             const int batch_start = start + j * n_batch;
             const int batch_size  = std::min(end - batch_start, n_batch);
 
+            llama_batch batch = llama_batch_init(batch_size, 0, 1);
```

Comment:
Move the `llama_batch` outside the loop and reuse it. Maybe utilize the `common_batch_` API to make it little less cumbersome.

Reply:
Fixed in https://github.com/ggerganov/llama.cpp/pull/9745/commits/734f9e29de8421afac198b86ad454937d94e672c and https://github.com/ggerganov/llama.cpp/pull/9745/commits/4be7ecf25e5a13c4a4f614f4b7d3bf095805c9ba